walletkit_client.go

```diff
@@ -737,28 +738,69 @@ func (m *walletKitClient) ListSweepsVerbose(ctx context.Context,
 	return result, nil
 }
 
+// BumpFeeOption customizes a BumpFee call.
+type BumpFeeOption func(*walletrpc.BumpFeeRequest)
+
+// WithImmediate is an option for enabling the immediate mode of BumpFee. The
+// sweeper will sweep this input without waiting for the next batch.
+func WithImmediate() BumpFeeOption {
+	return func(r *walletrpc.BumpFeeRequest) {
+		r.Immediate = true
+	}
+}
+
+// WithTargetConf is an option for setting the target_conf of BumpFee. If set,
+// the underlying fee estimator will use the target_conf to estimate the
+// starting fee rate for the fee function. The value of feeRate passed to
+// BumpFee will be ignored.
+func WithTargetConf(targetConf uint32) BumpFeeOption {
+	return func(r *walletrpc.BumpFeeRequest) {
+		r.TargetConf = targetConf
+
+		// We unset sat_per_vbyte, because if it is specified, it is
+		// used instead of target_conf.
+		r.SatPerVbyte = 0
+	}
+}
+
+// WithBudget is an option for setting the budget of BumpFee. It is the max
+// amount in sats that can be used as the fees. Setting this value greater than
+// the input's value may result in CPFP - one or more wallet utxos will be used
+// to pay the fees specified by the budget. If not set, for new inputs, by
+// default 50% of the input's value will be treated as the budget for fee
+// bumping; for existing inputs, their current budgets will be retained.
```

Comment:
maybe just let the descripiton of the RPC speak for itself and do not require these extensive comments ?

Reply:
I copy-pasted this from https://lightning.engineering/api-docs/api/lnd/wallet-kit/bump-fee/
All parts of this description seem to be important since they define the behavior. IMHO better to keep this in the documentation as well so it is easier to access this information.

Reply:
also specify that since 18.5 if the budget is set, the deadline has to be set too.

Reply:
I was a bit confused, if you select the deadline you need to specify the budget, not the other way around.

https://github.com/lightningnetwork/lnd/blob/68105be1ebbd8cae18c74f112d85382702477d80/lnrpc/walletrpc/walletkit_server.go#L1022-L1023

Reply:
yes, it would be cool if you'd open a PR.

Reply:
In this version there is no deadline setting, since it is not supported by 0.18.4 anyway.
We can add the comment when we add the deadline setting (for lndclient 0.18.5+).